src/main/java/backend/resource/MultiModel.java

```diff
@@ -49,6 +49,14 @@ private synchronized MultiModel add(Model model) {
         return this;
     }
 
+    public synchronized MultiModel removeRepoModelById(String repoId) {
+        Optional<Model> repoModelToBeDeleted = getModelById(repoId);
+        if (repoModelToBeDeleted.isPresent()) {
+            this.models.remove(repoModelToBeDeleted.get().getRepoId());
+        }
```

Comment:
I guess you could log if the repo doesn't exist, since that's probably indicative of a bug.